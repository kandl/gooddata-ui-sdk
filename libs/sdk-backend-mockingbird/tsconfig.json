{
    "compilerOptions": {
        "target": "es2017",
        "module": "NodeNext",
        "moduleResolution": "nodenext",
        "esModuleInterop": true,
        "lib": ["es2017", "dom"],
        "importHelpers": true,
        "sourceMap": true,
        "outDir": "./esm",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "baseUrl": "..",
        "paths": {
            "@gooddata/sdk-model": ["sdk-model/src"],
            "@gooddata/sdk-backend-base": ["sdk-backend-base/src"],
            "@gooddata/sdk-backend-spi": ["sdk-backend-spi/src"]
        }
    },
    "exclude": ["**/node_modules", "esm"]
}
