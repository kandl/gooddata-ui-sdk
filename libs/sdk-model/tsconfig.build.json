{
    "extends": "./tsconfig.json",
    "include": ["src/index.ts", "src/internal.ts"],
    "compilerOptions": {
        "baseUrl": null,
        "paths": null,
        "incremental": true,
        "declaration": true,
        "declarationMap": true
    }
}
