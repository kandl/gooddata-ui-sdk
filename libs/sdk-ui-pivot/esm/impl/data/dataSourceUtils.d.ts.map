{"version":3,"file":"dataSourceUtils.d.ts","sourceRoot":"","sources":["../../../src/impl/data/dataSourceUtils.ts"],"names":[],"mappings":"AACA,OAAO,EAAE,MAAM,EAAE,OAAO,EAAE,MAAM,gCAAgC,CAAC;AAGjE,OAAO,EAAE,cAAc,EAAE,MAAM,kBAAkB,CAAC;AAElD,OAAO,EAAE,UAAU,EAAE,MAAM,qBAAqB,CAAC;AAEjD,wBAAgB,gBAAgB,CAAC,OAAO,EAAE,OAAO,GAAG,SAAS,EAAE,SAAS,EAAE,cAAc,EAAE,GAAG,IAAI,GAAG,OAAO,CAgB1G;AAED,wBAAgB,uBAAuB,CAAC,QAAQ,EAAE,MAAM,EAAE,OAAO,EAAE,OAAO,GAAG,SAAS,GAAG,OAAO,CAO/F;AAED,wBAAgB,WAAW,CAAC,OAAO,EAAE,MAAM,GAAG,SAAS,GAAG,OAAO,CAIhE;AAED,wBAAgB,iBAAiB,CAAC,MAAM,EAAE,MAAM,GAAG,OAAO,CAEzD;AAED,wBAAgB,aAAa,CAAC,MAAM,EAAE,MAAM,GAAG,OAAO,CAIrD;AAED,wBAAgB,gBAAgB,CAAC,MAAM,EAAE,MAAM,GAAG,OAAO,CAIxD;AAED,wBAAgB,iBAAiB,CAAC,SAAS,EAAE,UAAU,GAAG,SAAS,GAAG,CAAC,MAAM,GAAG,IAAI,CAAC,EAAE,CAoBtF"}