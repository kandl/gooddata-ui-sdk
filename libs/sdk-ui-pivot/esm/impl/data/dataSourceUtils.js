import ApiWrapper from "../base/agApiWrapper.js";
import isEqual from "lodash/isEqual.js";
import { ROW_SUBTOTAL, ROW_TOTAL, COLUMN_SUBTOTAL, COLUMN_TOTAL } from "../base/constants.js";
export function areTotalsChanged(gridApi, newTotals) {
    var _a, _b, _c;
    const totals = newTotals !== null && newTotals !== void 0 ? newTotals : [];
    const currentTotalsCount = (_a = gridApi === null || gridApi === void 0 ? void 0 : gridApi.getPinnedBottomRowCount()) !== null && _a !== void 0 ? _a : 0;
    const newTotalsCount = (_b = newTotals === null || newTotals === void 0 ? void 0 : newTotals.length) !== null && _b !== void 0 ? _b : 0;
    if (currentTotalsCount !== newTotalsCount) {
        return true;
    }
    for (let i = 0; i < currentTotalsCount; i++) {
        if (!isEqual((_c = gridApi === null || gridApi === void 0 ? void 0 : gridApi.getPinnedBottomRow(i)) === null || _c === void 0 ? void 0 : _c.data, totals[i])) {
            return true;
        }
    }
    return false;
}
export function isInvalidGetRowsRequest(startRow, gridApi) {
    const bottomRowIndex = gridApi ? ApiWrapper.getPaginationBottomRowIndex(gridApi) : null;
    if (bottomRowIndex !== null) {
        return startRow > bottomRowIndex;
    }
    return false;
}
export function isSomeTotal(rowType) {
    const isRowTotal = rowType === ROW_TOTAL;
    const isRowSubtotal = rowType === ROW_SUBTOTAL;
    return isRowTotal || isRowSubtotal;
}
export function isSomeColumnTotal(colDef) {
    return isColumnTotal(colDef) || isColumnSubtotal(colDef);
}
export function isColumnTotal(colDef) {
    const { type } = colDef;
    return type === COLUMN_TOTAL;
}
export function isColumnSubtotal(colDef) {
    const { type } = colDef;
    return type === COLUMN_SUBTOTAL;
}
export function getSubtotalStyles(dimension) {
    if (!(dimension === null || dimension === void 0 ? void 0 : dimension.totals)) {
        return [];
    }
    let even = false;
    const subtotalStyles = dimension.itemIdentifiers.slice(1).map((attributeIdentifier) => {
        var _a, _b;
        const hasSubtotal = (_b = (_a = dimension.totals) === null || _a === void 0 ? void 0 : _a.some((total) => total.attributeIdentifier === attributeIdentifier)) !== null && _b !== void 0 ? _b : false;
        if (hasSubtotal) {
            even = !even;
            return even ? "even" : "odd";
        }
        return null;
    });
    // Grand total (first) has no styles
    return [null, ...subtotalStyles];
}
//# sourceMappingURL=dataSourceUtils.js.map