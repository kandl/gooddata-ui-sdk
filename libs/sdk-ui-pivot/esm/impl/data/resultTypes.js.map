{"version":3,"file":"resultTypes.js","sourceRoot":"","sources":["../../../src/impl/data/resultTypes.ts"],"names":[],"mappings":"AAEA,OAAO,EAAE,SAAS,EAAE,MAAM,sBAAsB,CAAC;AA+DjD,MAAM,UAAU,eAAe,CAAC,GAAY;IACxC,OAAO,CAAC,GAAsB,aAAtB,GAAG,uBAAH,GAAG,CAAqB,IAAI,MAAK,SAAS,CAAC;AACvD,CAAC"}