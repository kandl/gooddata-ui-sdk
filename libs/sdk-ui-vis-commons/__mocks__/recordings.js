// (C) 2020 GoodData Corporation
import { DataViewFacade } from "@gooddata/sdk-ui";
import { recordedDataView } from "@gooddata/sdk-backend-mockingbird";
export function recordedDataFacade(rec) {
    return DataViewFacade.for(recordedDataView(rec));
}
//# sourceMappingURL=recordings.js.map