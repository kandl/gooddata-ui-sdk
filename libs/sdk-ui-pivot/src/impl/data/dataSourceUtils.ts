// (C) 2019-2023 GoodData Corporation
import { GridApi } from "@ag-grid-community/all-modules";
import ApiWrapper from "../base/agApiWrapper.js";
import isEqual from "lodash/isEqual.js";
import { IGridTotalsRow } from "./resultTypes.js";
import { ROW_SUBTOTAL, ROW_TOTAL } from "../base/constants.js";
import { IDimension } from "@gooddata/sdk-model";

export function areTotalsChanged(gridApi: GridApi | undefined, newTotals: IGridTotalsRow[] | null): boolean {
    const totals = newTotals ?? [];
    const currentTotalsCount = gridApi?.getPinnedBottomRowCount() ?? 0;
    const newTotalsCount = newTotals?.length ?? 0;

    if (currentTotalsCount !== newTotalsCount) {
        return true;
    }

    for (let i = 0; i < currentTotalsCount; i++) {
        if (!isEqual(gridApi?.getPinnedBottomRow(i)?.data, totals[i])) {
            return true;
        }
    }

    return false;
}

export function isInvalidGetRowsRequest(startRow: number, gridApi: GridApi | undefined): boolean {
    const bottomRowIndex = gridApi ? ApiWrapper.getPaginationBottomRowIndex(gridApi) : null;
    if (bottomRowIndex !== null) {
        return startRow > bottomRowIndex;
    }

    return false;
}

export function isSomeTotal(rowType: string | undefined): boolean {
    const isRowTotal = rowType === ROW_TOTAL;
    const isRowSubtotal = rowType === ROW_SUBTOTAL;
    return isRowTotal || isRowSubtotal;
}

export function getSubtotalStyles(dimension: IDimension | undefined): (string | null)[] {
    if (!dimension?.totals) {
        return [];
    }

    let even = false;
    const subtotalStyles = dimension.itemIdentifiers.slice(1).map((attributeIdentifier) => {
        const hasSubtotal =
            dimension.totals?.some((total) => total.attributeIdentifier === attributeIdentifier) ?? false;

        if (hasSubtotal) {
            even = !even;
            return even ? "even" : "odd";
        }

        return null;
    });

    // Grand total (first) has no styles
    return [null, ...subtotalStyles];
}
