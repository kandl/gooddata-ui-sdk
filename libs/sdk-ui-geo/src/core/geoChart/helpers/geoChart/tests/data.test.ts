// (C) 2020-2023 GoodData Corporation
import { getGeoData, getLocation, parseCoordinate } from "../data.js";
import { dummyDataView } from "@gooddata/sdk-backend-mockingbird";
import { emptyDef } from "@gooddata/sdk-model";
import { DataViewFacade } from "@gooddata/sdk-ui";
import { RecShortcuts } from "../../../../../../__mocks__/recordings.js";

describe("getLocation", () => {
    it("should return { lat, lng } from location string", () => {
        const location = getLocation("44.500000;-89.500000");
        expect(location).toEqual({
            lat: 44.5,
            lng: -89.5,
        });
    });

    it.each(["", "123"])("should return null when input is '%s'", (input: string) => {
        const location = getLocation(input);
        expect(location).toEqual(null);
    });
});

describe("parseCoordinate", () => {
    it.each([
        ["44.500000", 44.5],
        ["-89.500000", -89.5],
    ])("should return number from coordinate string '%s'", (input: string, result: number) => {
        const location = parseCoordinate(input);
        expect(location).toEqual(result);
    });

    it.each(["abc", null])("should return null for invalid input '%s'", (input: string | null) => {
        const location = parseCoordinate(input);
        expect(location).toEqual(null);
    });
});

describe("geoChartData", () => {
    it("should return geoData with empty bucket", () => {
        const emptyDv = DataViewFacade.for(dummyDataView(emptyDef("testWorkspace")));

        const geoData = getGeoData(emptyDv, "empty value", "null value");
        expect(geoData).toEqual({});
    });

    it("should return geoData with all buckets", () => {
        const { geoData } = RecShortcuts.AllAndSmall;

        expect(geoData).toMatchSnapshot();
    });

    it("should return geoData with location, tooltipText, size", () => {
        const { geoData } = RecShortcuts.LocationSizeAndTooltip_Small;
        expect(geoData).toMatchSnapshot();
    });

    it("should return geoData with location, color, size", () => {
        const { geoData } = RecShortcuts.LocationSizeAndColor_Small;
        expect(geoData).toMatchSnapshot();
    });

    it("should return geoData with latitude, longitude and tooltipText", () => {
        const { geoData } = RecShortcuts.LatitudeAndLongitudeOnlyWithTooltip;
        expect(geoData).toMatchSnapshot();
    });
});
