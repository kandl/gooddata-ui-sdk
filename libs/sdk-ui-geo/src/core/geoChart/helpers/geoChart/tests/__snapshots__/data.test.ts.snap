// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`geoChartData > should return geoData with all buckets 1`] = `
{
  "color": {
    "data": [
      4969,
      256,
      56,
      0,
      1856,
      1305,
      628,
      982,
      1683,
      3295,
      7286,
    ],
    "format": "#,##0.00",
    "index": 1,
    "name": "Density",
  },
  "location": {
    "data": [
      {
        "lat": 25.7839,
        "lng": -80.2102,
      },
      {
        "lat": 26.2181,
        "lng": -98.418,
      },
      {
        "lat": 26.2585,
        "lng": -97.695,
      },
      {
        "lat": 26.3024,
        "lng": -98.6425,
      },
      {
        "lat": 26.5281,
        "lng": -80.0811,
      },
      {
        "lat": 27.9942,
        "lng": -82.4451,
      },
      {
        "lat": 28.2405,
        "lng": -82.1796,
      },
      {
        "lat": 28.4772,
        "lng": -81.3369,
      },
      {
        "lat": 32.8312,
        "lng": -117.1225,
      },
      {
        "lat": 34.1139,
        "lng": -118.4068,
      },
      {
        "lat": 37.7562,
        "lng": -122.443,
      },
    ],
    "index": 0,
    "name": "City",
  },
  "segment": {
    "data": [
      "Florida",
      "Texas",
      "Texas",
      "Texas",
      "Florida",
      "Florida",
      "Florida",
      "Florida",
      "California",
      "California",
      "California",
    ],
    "index": 1,
    "name": "State Name",
    "uris": [
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340105",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=4436534",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=4436534",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=4436534",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340105",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340105",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340105",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340105",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340116",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340116",
      "/gdc/md/liveexamplesworkspace/obj/9461/elements?id=6340116",
    ],
  },
  "size": {
    "data": [
      6381966,
      2228,
      121,
      0,
      77702,
      2804240,
      76416,
      1776841,
      3210314,
      1.28,
      3603761,
    ],
    "format": "#,##0.00",
    "index": 0,
    "name": "Population",
  },
  "tooltipText": {
    "data": [
      "Miami",
      "Abram",
      "Lasana",
      "Longoria",
      "Boynton Beach",
      "Tampa",
      "Zephyrhills",
      "Orlando",
      "San Diego",
      "Los Angeles",
      "San Francisco",
    ],
    "index": 2,
    "name": "City",
  },
}
`;

exports[`geoChartData > should return geoData with latitude, longitude and tooltipText 1`] = `
{
  "latitude": {
    "index": 0,
    "name": "City",
  },
  "location": {
    "data": [
      {
        "lat": 18.0722,
        "lng": 18.0722,
      },
      {
        "lat": 18.0727,
        "lng": 18.0727,
      },
      {
        "lat": 18.0729,
        "lng": 18.0729,
      },
      {
        "lat": 21.2975,
        "lng": 21.2975,
      },
      {
        "lat": 21.3294,
        "lng": 21.3294,
      },
      {
        "lat": 21.4031,
        "lng": 21.4031,
      },
      {
        "lat": 25.7839,
        "lng": 25.7839,
      },
      {
        "lat": 26.2181,
        "lng": 26.2181,
      },
      {
        "lat": 26.2585,
        "lng": 26.2585,
      },
      {
        "lat": 26.3024,
        "lng": 26.3024,
      },
      {
        "lat": 26.5281,
        "lng": 26.5281,
      },
      {
        "lat": 27.9942,
        "lng": 27.9942,
      },
      {
        "lat": 28.2405,
        "lng": 28.2405,
      },
      {
        "lat": 28.4772,
        "lng": 28.4772,
      },
      {
        "lat": 29.3443,
        "lng": 29.3443,
      },
      {
        "lat": 29.8781,
        "lng": 29.8781,
      },
      {
        "lat": 30.006,
        "lng": 30.006,
      },
      {
        "lat": 30.2492,
        "lng": 30.2492,
      },
      {
        "lat": 30.6782,
        "lng": 30.6782,
      },
      {
        "lat": 30.8193,
        "lng": 30.8193,
      },
      {
        "lat": 31.1545,
        "lng": 31.1545,
      },
      {
        "lat": 31.8194,
        "lng": 31.8194,
      },
      {
        "lat": 32.0281,
        "lng": 32.0281,
      },
      {
        "lat": 32.1061,
        "lng": 32.1061,
      },
      {
        "lat": 32.1545,
        "lng": 32.1545,
      },
      {
        "lat": 32.6886,
        "lng": 32.6886,
      },
      {
        "lat": 32.8312,
        "lng": 32.8312,
      },
      {
        "lat": 33.3645,
        "lng": 33.3645,
      },
      {
        "lat": 33.4017,
        "lng": 33.4017,
      },
      {
        "lat": 33.5277,
        "lng": 33.5277,
      },
      {
        "lat": 33.5304,
        "lng": 33.5304,
      },
      {
        "lat": 33.5722,
        "lng": 33.5722,
      },
      {
        "lat": 33.6514,
        "lng": 33.6514,
      },
      {
        "lat": 33.6933,
        "lng": 33.6933,
      },
      {
        "lat": 33.7935,
        "lng": 33.7935,
      },
      {
        "lat": 33.8945,
        "lng": 33.8945,
      },
      {
        "lat": 33.9687,
        "lng": 33.9687,
      },
      {
        "lat": 34.1139,
        "lng": 34.1139,
      },
      {
        "lat": 34.3594,
        "lng": 34.3594,
      },
      {
        "lat": 34.5564,
        "lng": 34.5564,
      },
      {
        "lat": 34.6152,
        "lng": 34.6152,
      },
      {
        "lat": 34.6288,
        "lng": 34.6288,
      },
      {
        "lat": 34.6988,
        "lng": 34.6988,
      },
      {
        "lat": 34.8148,
        "lng": 34.8148,
      },
      {
        "lat": 34.9944,
        "lng": 34.9944,
      },
      {
        "lat": 34.9952,
        "lng": 34.9952,
      },
      {
        "lat": 35.1629,
        "lng": 35.1629,
      },
      {
        "lat": 35.448,
        "lng": 35.448,
      },
      {
        "lat": 36.0404,
        "lng": 36.0404,
      },
      {
        "lat": 36.1844,
        "lng": 36.1844,
      },
      {
        "lat": 36.3713,
        "lng": 36.3713,
      },
      {
        "lat": 36.5055,
        "lng": 36.5055,
      },
      {
        "lat": 36.5463,
        "lng": 36.5463,
      },
      {
        "lat": 36.5879,
        "lng": 36.5879,
      },
      {
        "lat": 36.6844,
        "lng": 36.6844,
      },
      {
        "lat": 36.7388,
        "lng": 36.7388,
      },
      {
        "lat": 36.7596,
        "lng": 36.7596,
      },
      {
        "lat": 36.8428,
        "lng": 36.8428,
      },
      {
        "lat": 36.8432,
        "lng": 36.8432,
      },
      {
        "lat": 36.9352,
        "lng": 36.9352,
      },
      {
        "lat": 37.0018,
        "lng": 37.0018,
      },
      {
        "lat": 37.0171,
        "lng": 37.0171,
      },
      {
        "lat": 37.1674,
        "lng": 37.1674,
      },
      {
        "lat": 37.2563,
        "lng": 37.2563,
      },
      {
        "lat": 37.4788,
        "lng": 37.4788,
      },
      {
        "lat": 37.6213,
        "lng": 37.6213,
      },
      {
        "lat": 37.7562,
        "lng": 37.7562,
      },
      {
        "lat": 37.805,
        "lng": 37.805,
      },
      {
        "lat": 37.9748,
        "lng": 37.9748,
      },
      {
        "lat": 38.0054,
        "lng": 38.0054,
      },
      {
        "lat": 38.2458,
        "lng": 38.2458,
      },
      {
        "lat": 38.4099,
        "lng": 38.4099,
      },
      {
        "lat": 38.5352,
        "lng": 38.5352,
      },
      {
        "lat": 38.6353,
        "lng": 38.6353,
      },
      {
        "lat": 38.6623,
        "lng": 38.6623,
      },
      {
        "lat": 38.9047,
        "lng": 38.9047,
      },
      {
        "lat": 38.9372,
        "lng": 38.9372,
      },
      {
        "lat": 39.0127,
        "lng": 39.0127,
      },
      {
        "lat": 39.0256,
        "lng": 39.0256,
      },
      {
        "lat": 39.0699,
        "lng": 39.0699,
      },
      {
        "lat": 39.1284,
        "lng": 39.1284,
      },
      {
        "lat": 39.181,
        "lng": 39.181,
      },
      {
        "lat": 39.4151,
        "lng": 39.4151,
      },
      {
        "lat": 39.5939,
        "lng": 39.5939,
      },
      {
        "lat": 39.6152,
        "lng": 39.6152,
      },
      {
        "lat": 39.7243,
        "lng": 39.7243,
      },
      {
        "lat": 39.7255,
        "lng": 39.7255,
      },
      {
        "lat": 39.7264,
        "lng": 39.7264,
      },
      {
        "lat": 39.8403,
        "lng": 39.8403,
      },
      {
        "lat": 39.9384,
        "lng": 39.9384,
      },
      {
        "lat": 40.0048,
        "lng": 40.0048,
      },
      {
        "lat": 40.3232,
        "lng": 40.3232,
      },
      {
        "lat": 40.4394,
        "lng": 40.4394,
      },
      {
        "lat": 40.5887,
        "lng": 40.5887,
      },
      {
        "lat": 40.6207,
        "lng": 40.6207,
      },
      {
        "lat": 40.6943,
        "lng": 40.6943,
      },
      {
        "lat": 40.7467,
        "lng": 40.7467,
      },
      {
        "lat": 41.0056,
        "lng": 41.0056,
      },
      {
        "lat": 41.208,
        "lng": 41.208,
      },
      {
        "lat": 41.2636,
        "lng": 41.2636,
      },
      {
        "lat": 41.2739,
        "lng": 41.2739,
      },
      {
        "lat": 41.4625,
        "lng": 41.4625,
      },
      {
        "lat": 41.5367,
        "lng": 41.5367,
      },
      {
        "lat": 41.5725,
        "lng": 41.5725,
      },
      {
        "lat": 41.6698,
        "lng": 41.6698,
      },
      {
        "lat": 41.6812,
        "lng": 41.6812,
      },
      {
        "lat": 41.7226,
        "lng": 41.7226,
      },
      {
        "lat": 41.7948,
        "lng": 41.7948,
      },
      {
        "lat": 41.8083,
        "lng": 41.8083,
      },
      {
        "lat": 41.8864,
        "lng": 41.8864,
      },
      {
        "lat": 41.9234,
        "lng": 41.9234,
      },
      {
        "lat": 41.9311,
        "lng": 41.9311,
      },
      {
        "lat": 41.9592,
        "lng": 41.9592,
      },
      {
        "lat": 42.001,
        "lng": 42.001,
      },
      {
        "lat": 42.0014,
        "lng": 42.0014,
      },
      {
        "lat": 42.0133,
        "lng": 42.0133,
      },
      {
        "lat": 42.0747,
        "lng": 42.0747,
      },
      {
        "lat": 42.1057,
        "lng": 42.1057,
      },
      {
        "lat": 42.1638,
        "lng": 42.1638,
      },
      {
        "lat": 42.4919,
        "lng": 42.4919,
      },
      {
        "lat": 42.5801,
        "lng": 42.5801,
      },
      {
        "lat": 42.6407,
        "lng": 42.6407,
      },
      {
        "lat": 42.7491,
        "lng": 42.7491,
      },
      {
        "lat": 42.8803,
        "lng": 42.8803,
      },
      {
        "lat": 42.8848,
        "lng": 42.8848,
      },
      {
        "lat": 43.0145,
        "lng": 43.0145,
      },
      {
        "lat": 43.0883,
        "lng": 43.0883,
      },
      {
        "lat": 43.1,
        "lng": 43.1,
      },
      {
        "lat": 43.182,
        "lng": 43.182,
      },
      {
        "lat": 43.4244,
        "lng": 43.4244,
      },
      {
        "lat": 43.4865,
        "lng": 43.4865,
      },
      {
        "lat": 43.5053,
        "lng": 43.5053,
      },
      {
        "lat": 43.5724,
        "lng": 43.5724,
      },
      {
        "lat": 43.5936,
        "lng": 43.5936,
      },
      {
        "lat": 43.7348,
        "lng": 43.7348,
      },
      {
        "lat": 43.857,
        "lng": 43.857,
      },
      {
        "lat": 43.9106,
        "lng": 43.9106,
      },
      {
        "lat": 44.2098,
        "lng": 44.2098,
      },
      {
        "lat": 44.6384,
        "lng": 44.6384,
      },
      {
        "lat": 44.9076,
        "lng": 44.9076,
      },
      {
        "lat": 44.9083,
        "lng": 44.9083,
      },
      {
        "lat": 44.9906,
        "lng": 44.9906,
      },
      {
        "lat": 45.3436,
        "lng": 45.3436,
      },
      {
        "lat": 45.5812,
        "lng": 45.5812,
      },
      {
        "lat": 45.7099,
        "lng": 45.7099,
      },
      {
        "lat": 45.7796,
        "lng": 45.7796,
      },
      {
        "lat": 45.9425,
        "lng": 45.9425,
      },
      {
        "lat": 46.3004,
        "lng": 46.3004,
      },
      {
        "lat": 46.4866,
        "lng": 46.4866,
      },
      {
        "lat": 46.812,
        "lng": 46.812,
      },
      {
        "lat": 47.4982,
        "lng": 47.4982,
      },
      {
        "lat": 57.2401,
        "lng": 57.2401,
      },
      {
        "lat": 61.1508,
        "lng": 61.1508,
      },
      {
        "lat": 61.6053,
        "lng": 61.6053,
      },
    ],
    "index": 0,
    "name": "City",
  },
  "longitude": {
    "index": 1,
    "name": "City",
  },
  "tooltipText": {
    "data": [
      "Santo Domingo",
      "La Yuca",
      "Martorell",
      "East Honolulu",
      "Honolulu",
      "Pearl City",
      "Miami",
      "Abram",
      "Lasana",
      "Longoria",
      "Boynton Beach",
      "Tampa",
      "Zephyrhills",
      "Orlando",
      "Triumph",
      "Timberlane",
      "Kaplan",
      "Pearlington",
      "Mobile",
      "Kingsland",
      "Leakesville",
      "Sunland Park",
      "Savannah",
      "Utica",
      "Tucson",
      "Virden",
      "San Diego",
      "Augusta",
      "Mesa",
      "Birmingham",
      "Aiken",
      "Phoenix",
      "Lower Frisco",
      "Manning",
      "Thackerville",
      "Calabash",
      "Willington",
      "Los Angeles",
      "Comanche",
      "Sneads Ferry",
      "Bryant",
      "Roff",
      "Huntsville",
      "McFarlan",
      "Lookout Mountain",
      "Copperhill",
      "Cowan",
      "Van Buren",
      "Hornersville",
      "Siloam Springs",
      "Berryville",
      "Hazel",
      "Virgilina",
      "New Madrid",
      "Baskerville",
      "Goodman",
      "Columbus",
      "Ivanhoe",
      "Harlan",
      "Beatty",
      "Waldron",
      "Hildale",
      "Oswego",
      "Montezuma Creek",
      "Mulvane",
      "Blanding",
      "San Francisco",
      "Rossmore",
      "Smith Island",
      "Big Creek",
      "Marmet",
      "Burlington",
      "La Plata",
      "Blades",
      "Cedarville",
      "Washington",
      "Cape May Point",
      "Felton",
      "Genoa",
      "Fruit Hill",
      "Rodney Village",
      "Newtonsville",
      "Fallon Station",
      "Collings Lakes",
      "Greenwood Village",
      "Pen Mar",
      "Blue Ridge Summit",
      "Brave",
      "Welby",
      "Cherry Hill Mall",
      "Byron",
      "Bladen",
      "Holdrege",
      "Long Beach",
      "Sterling",
      "New York",
      "Wyandanch",
      "Washam",
      "Hillsdale",
      "Siasconset",
      "West Haven",
      "Carolina",
      "Meriden",
      "Des Moines",
      "North Rock Springs",
      "Bristol",
      "Morenci",
      "Westmont",
      "North Aurora",
      "Villa Park",
      "Valley Falls",
      "Attleboro",
      "Colon",
      "Woonsocket",
      "New Pine Creek",
      "Malin",
      "Green Harbor-Cedar Crest",
      "Brooklyn",
      "Vinton",
      "Dakota Dunes",
      "Gratiot",
      "Springfield",
      "Nashua",
      "Oak Creek",
      "Old Bennington",
      "Fort Hall",
      "Kittery Point",
      "Garner",
      "Oelrichs",
      "Sanford",
      "Monroe",
      "Bigelow",
      "Laconia",
      "North Hartland",
      "Falmouth Foreside",
      "Harper",
      "Dakota",
      "Sanborn",
      "Lima",
      "McCall",
      "Enosburg Falls",
      "West Stewartstown",
      "Roberts",
      "Washougal",
      "Salmon Creek",
      "Belgrade",
      "Forbes",
      "Skamokawa Valley",
      "Orofino",
      "Glen Ullin",
      "Mayville",
      "Sitka",
      "Anchorage",
      "Lakes",
    ],
    "index": 2,
    "name": "City",
  },
}
`;

exports[`geoChartData > should return geoData with location, color, size 1`] = `
{
  "color": {
    "data": [
      698,
      2145,
      952,
      803,
      2234,
      1960,
      4969,
      256,
      56,
      0,
    ],
    "format": "#,##0.00",
    "index": 1,
    "name": "Density",
  },
  "location": {
    "data": [
      {
        "lat": 18.0722,
        "lng": -66.7458,
      },
      {
        "lat": 18.0727,
        "lng": -66.6077,
      },
      {
        "lat": 18.0729,
        "lng": -65.8976,
      },
      {
        "lat": 21.2975,
        "lng": -157.7211,
      },
      {
        "lat": 21.3294,
        "lng": -157.846,
      },
      {
        "lat": 21.4031,
        "lng": -157.9566,
      },
      {
        "lat": 25.7839,
        "lng": -80.2102,
      },
      {
        "lat": 26.2181,
        "lng": -98.418,
      },
      {
        "lat": 26.2585,
        "lng": -97.695,
      },
      {
        "lat": 26.3024,
        "lng": -98.6425,
      },
    ],
    "index": 0,
    "name": "City",
  },
  "size": {
    "data": [
      3868,
      525,
      2254,
      47868,
      833671,
      46259,
      6381966,
      2228,
      121,
      0,
    ],
    "format": "#,##0.00",
    "index": 0,
    "name": "Population",
  },
}
`;

exports[`geoChartData > should return geoData with location, tooltipText, size 1`] = `
{
  "location": {
    "data": [
      {
        "lat": 18.0722,
        "lng": -66.7458,
      },
      {
        "lat": 18.0727,
        "lng": -66.6077,
      },
      {
        "lat": 18.0729,
        "lng": -65.8976,
      },
      {
        "lat": 21.2975,
        "lng": -157.7211,
      },
      {
        "lat": 21.3294,
        "lng": -157.846,
      },
      {
        "lat": 21.4031,
        "lng": -157.9566,
      },
      {
        "lat": 25.7839,
        "lng": -80.2102,
      },
      {
        "lat": 26.2181,
        "lng": -98.418,
      },
      {
        "lat": 26.2585,
        "lng": -97.695,
      },
      {
        "lat": 26.3024,
        "lng": -98.6425,
      },
    ],
    "index": 0,
    "name": "City",
  },
  "size": {
    "data": [
      3868,
      525,
      2254,
      47868,
      833671,
      46259,
      6381966,
      2228,
      121,
      0,
    ],
    "format": "#,##0.00",
    "index": 0,
    "name": "Population",
  },
  "tooltipText": {
    "data": [
      "Santo Domingo",
      "La Yuca",
      "Martorell",
      "East Honolulu",
      "Honolulu",
      "Pearl City",
      "Miami",
      "Abram",
      "Lasana",
      "Longoria",
    ],
    "index": 1,
    "name": "City",
  },
}
`;
